### PR TITLE
feat(prefect-aws): Add AWS AssumeRole support to AWSCredentials Block

### LINKED ISSUE #19831: Support for AWS IAM Assume Role authentication in AWSCredentials Block
### Describe the current behavior

The [`AWSCredentials`](https://docs.prefect.io/integrations/prefect-aws/index#credentials) block does not support AWS IAM Assume Role functionality.

Having AssumeRole support would be a big win because it matches AWS best practices: short-lived, automatically rotated credentials; least privilege via role scoping; and clean separation between “who I am” and “what I can do” across accounts and environments. In fact, this is the second recommendation under [AWS's security best practices for IAM](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#bp-workloads-use-roles).

It would let Prefect users model the full auth chain declaratively in one place (the block), making deployments more reproducible, reducing secret sprawl, and enabling common enterprise setups (cross-account S3, Secrets Manager, ECS, etc.) without bespoke glue code or environment-specific hacks.

### Describe the proposed behavior

Add two new optional fields to the current `AWSCredentials` Block:

- `role_arn` - the role to assume
- `assume_role_kwargs` - these are passed to the Boto3 STS Client [`assume_role` method](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sts/client/assume_role.html) call. For example, many AWS IAM roles have an `ExternalID` defined to prevent the [confused deputy problem](https://docs.aws.amazon.com/IAM/latest/UserGuide/confused-deputy.html).

If `role_arn` is set, take the base set of credentials defined and apply them to the `assume_role` invocation. Return the resulting temporary credentials.

Current behavior does not change and the block is fully backwards compatible.

### Example Use

Use an instance role's credentials to assume another role:

```python
aws_credentials = AwsCredentials(
    role_arn="arn:aws:iam::123456789012:role/MyRole"
)
s3_client = aws_credentials.get_boto3_session().client("s3")
```

Customize the inputs to the `assume_role` call:

```python
aws_credentials = AwsCredentials(
    role_arn="arn:aws:iam::123456789012:role/MyRole",
    assume_role_kwargs={
        "RoleSessionName": "my-session",
        "DurationSeconds": 3600,
        "ExternalId": "unique-external-id"
    }
)
s3_client = aws_credentials.get_boto3_session().client("s3")
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bdalpe**
This PR adds support for assuming IAM roles in `AwsCredentials`, enabling cross-account access and temporary credentials.

### Summary

Adds `role_arn` and `assume_role_kwargs` to `AwsCredentials` to assume IAM roles via AWS STS. When `role_arn` is provided, the credentials block assumes the role and returns a session with temporary credentials.

### Changes

- **New `AssumeRoleParameters` model** (`assume_role_parameters.py`):
  - Pydantic model for assume role parameters
  - Supports: `RoleSessionName`, `DurationSeconds`, `Policy`, `PolicyArns`, `Tags`, `TransitiveTagKeys`, `ExternalId`, `SerialNumber`, `TokenCode`, `SourceIdentity`, `ProvidedContexts`
  - Includes `__hash__` and `get_params_override()` for parameter handling

- **Enhanced `AwsCredentials` class** (`credentials.py`):
  - Added `role_arn` field for the IAM role ARN to assume
  - Added `assume_role_kwargs` field (type `AssumeRoleParameters`) for additional assume role parameters
  - Updated `get_boto3_session()` to:
    - Detect `role_arn` and call STS `assume_role`
    - Generate a default `RoleSessionName` if not provided (format: `prefect-session-{uuid}`)
    - Pass all `assume_role_kwargs` to the STS call
    - Return a new boto3 session with temporary credentials from the assumed role
  - Updated `__hash__` to include `role_arn` and `assume_role_kwargs` for cache invalidation

- **Comprehensive test coverage** (`test_credentials.py`):
  - Basic assume role functionality
  - Custom `RoleSessionName` handling
  - Default `RoleSessionName` generation
  - Additional kwargs (e.g., `DurationSeconds`, `ExternalId`)
  - Hash behavior with role ARN and assume role kwargs
  - Nested structures (Tags, PolicyArns) in hash computation
  - Temporary credentials usage verification

### Usage Example

```python
from prefect_aws import AwsCredentials

# Basic assume role
credentials = AwsCredentials(
    role_arn="arn:aws:iam::123456789012:role/MyRole"
)

# With additional parameters
credentials = AwsCredentials(
    role_arn="arn:aws:iam::123456789012:role/MyRole",
    assume_role_kwargs={
        "RoleSessionName": "my-session",
        "DurationSeconds": 3600,
        "ExternalId": "unique-external-id"
    }
)

# Use the session with assumed role credentials
session = credentials.get_boto3_session()
s3_client = session.client("s3")
```

### Testing

All new functionality is covered by unit tests, including edge cases for parameter handling and hash computation.

### Backward Compatibility

Fully backward compatible. Existing `AwsCredentials` usage continues to work; assume role is only used when `role_arn` is provided.

Resolves #19831 

### Checklist

- [X] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: src/integrations/prefect-aws/prefect_aws/credentials.py [hunk 1]
```diff
@@ -107,6 +109,16 @@ class AwsCredentials(CredentialsBlock):
         description="Extra parameters to initialize the Client.",
         title="AWS Client Parameters",
     )
+    role_arn: Optional[str] = Field(
```
Comment (desertaxle) [inline src/integrations/prefect-aws/prefect_aws/credentials.py]: Could we name this `assume_role_arn` for some consistency between this field and `assume_role_kwargs`?
```suggestion
    assume_role_arn: Optional[str] = Field(
```

Comment (bdalpe) [reply src/integrations/prefect-aws/prefect_aws/credentials.py]: Yep, makes sense. I've updated for consistency.
